File: dev-docs/snippets/auth/magic-link-verify.swift
import JsBaoClient

// Verify a magic-link token and sign the user in via the typed `client.auth`
// namespace. Returns a typed `MagicLinkVerifyResult` (`user`, `isNewUser?`,
// `promptAddPasskey?`). On success the SDK has already applied the access token.
func magicLinkVerify(client: JsBaoClient, token: String) async throws {
  // #region example
  let result = try await client.auth.magicLinkVerify(token: token)
  let user = result.user           // AuthUser: userId, email, name?
  let isNewUser = result.isNewUser ?? false
  // #endregion example
  _ = (user, isNewUser)
}

File: dev-docs/snippets/auth/magic-link-verify.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Verify a magic-link token and sign the user in. `inviteToken` resolves
// deferred grants for invite-only signups.
export async function magicLinkVerify(
  client: JsBaoClient,
  token: string,
  inviteToken: string,
) {
  // #region example
  const { user, isNewUser, promptAddPasskey } = await client.magicLinkVerify(
    token,
    { inviteToken },
  );
  // #endregion example
  return { user, isNewUser, promptAddPasskey };
}
